packages/core/src/external/fhir/shared/bundle.ts

```diff
@@ -78,33 +80,60 @@ export function getReferencesFromResources({
   return { references, missingReferences };
 }
 
-function getReferencesFromRaw(
-  rawContents: string,
-  referencesToInclude: ResourceType[],
-  referencesToExclude: ResourceType[]
-): ReferenceWithIdAndType[] {
+// TODO 2355 Refactor this
+/**
+ * @deprecated This is not a proper implementation to return References. Those might be represented
+ * in different ways than the relative one "Patient/123". We should create a generic implementation
+ * based on `getPatientReferencesFromFhirBundle` so we can get all references from a list of
+ * resources (and update the patient's one to use it).
+ * @see https://github.com/metriport/metriport-internal/issues/2355
+ *
+ * Return the references found in the given resources.
+ *
+ * @param resources The resources to search for references
+ * @param resourceTypesToInclude The resource types to include in the result. If not set,
+ *        references with all resource types will be included.
+ * @param resourceTypesToExclude The resource types to exclude from the result. If not set,
+ *        no references will be excluded.
+ * @returns The references found in the given resources.
+ */
+export function getReferences({
+  resources,
+  referencesToInclude,
+  referencesToExclude = [],
+}: {
+  resources: Resource[] | undefined;
+  referencesToInclude?: ResourceType[] | undefined;
+  referencesToExclude?: ResourceType[] | undefined;
+}): ReferenceWithIdAndType[] {
+  if (!resources || resources.length <= 0) return [];
+  const rawContents = JSON.stringify(resources);
   const matches = rawContents.matchAll(referenceRegex);
-  const references = [];
+  const references: string[] = [];
   for (const match of matches) {
     const ref = match[1];
     if (ref) references.push(ref);
   }
   const uniqueRefs = uniq(references);
-  const preResult: ReferenceWithIdAndType[] = uniqueRefs.flatMap(r => {
-    const parts = r.split("/");
-    const type = parts[0] as ResourceType | undefined;
-    const id = parts[1];
-    if (!id || !type) return [];
-    return { type, id, reference: r };
-  });
-  if (referencesToInclude.length <= 0 && referencesToExclude.length <= 0) return preResult;
+  const preResult: ReferenceWithIdAndType[] = uniqueRefs
+    .flatMap(buildReferenceFromStringRelative)
+    .flatMap(filterTruthy);
+  if (!referencesToInclude && referencesToExclude.length <= 0) return preResult;
```

Comment:
I'd think about splitting 121 and 124 into blocks that first include then exclude... right now it's a confusing check IMO.

Reply:
done